auth0/src/test/java/com/auth0/android/authentication/AuthenticationAPIClientTest.kt

```diff
@@ -2772,6 +2738,283 @@ public class AuthenticationAPIClientTest {
         )
     }
 
+    //DPoP
+
+    @Test
+    public fun shouldNotAddDpopHeaderWhenDpopNotEnabled() {
+        mockAPI.willReturnSuccessfulLogin()
+        val callback = MockAuthenticationCallback<Credentials>()
+        // DPoP is not enabled - dPoP property should be null
+        client.login(SUPPORT_AUTH0_COM, PASSWORD, MY_CONNECTION)
+            .start(callback)
+        ShadowLooper.idleMainLooper()
+
+        val request = mockAPI.takeRequest()
+        assertThat(request.getHeader("DPoP"), Matchers.nullValue())
+        assertThat(request.path, Matchers.equalTo("/oauth/token"))
+        assertThat(
+            callback, AuthenticationCallbackMatcher.hasPayloadOfType(
+                Credentials::class.java
+            )
+        )
+    }
+
+    @Test
+    public fun shouldAddDpopHeaderWhenDpopEnabledAndKeyPairExists() {
+        whenever(mockKeyStore.hasKeyPair()).thenReturn(true)
+        whenever(mockKeyStore.getKeyPair()).thenReturn(Pair(FakeECPrivateKey(), FakeECPublicKey()))
+
+        mockAPI.willReturnSuccessfulLogin()
+        val callback = MockAuthenticationCallback<Credentials>()
+
+        // Enable DPoP
+        client.useDPoP().login(SUPPORT_AUTH0_COM, PASSWORD, MY_CONNECTION)
+            .start(callback)
+        ShadowLooper.idleMainLooper()
+
+        val request = mockAPI.takeRequest()
+        assertThat(request.getHeader("DPoP"), Matchers.notNullValue())
+        assertThat(request.path, Matchers.equalTo("/oauth/token"))
+        assertThat(
+            callback, AuthenticationCallbackMatcher.hasPayloadOfType(
+                Credentials::class.java
+            )
+        )
+    }
+
+    @Test
+    public fun shouldNotAddDpopHeaderWhenDpopEnabledButNoKeyPair() {
```

Comment:
For the login case (that is, requests to /token when the `grant_type` is not `refresh_token`), it should create a new key pair if one does not already exist.

Reply:
It should not create a key pair when the `grant_type` is `refresh_token`.